Update search API and change compendium type

nuest opened the pull request:
Deprecates #56

LukasLohoff commented:
A few questions:

- Will the query string (e.g. `q=10.3389/fpsyg.2013.00479`) be URL encoded?  I think we should also mention that in the docs

We are just using the standard analyzer, correct? So special characters aren't supported anyway. 
https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-analyzers.html

Right now both https://o2r.uni-muenster.de/api/v1/search?q=10.3389/fpsyg.2013.00479 and https://o2r.uni-muenster.de/api/v1/search?q=103389%2Ffpsyg201300479 return an error ("query_shard_exception")

- The example `http://o2r.uni-muenster.de/api/v1/search?q=spatial` might be a bit confusing because we are using "spatial" as the query string while it is also a type of query we will be supporting

- from https://github.com/o2r-project/o2r-web-api/pull/44#issuecomment-334722298: I think an example for spatial search / spatial+temporal search would be helpful. #44 has a spatial query as a query string, #57 switched to the Query DSL style, but has just a temporal query

nuest commented:
- removed the "spatial" query string example
- added some issues to finder
  - https://github.com/o2r-project/o2r-finder/issues/11
  - https://github.com/o2r-project/o2r-finder/issues/12
- added spatial search example
- good point on the analyzer, we'll look into that: https://github.com/o2r-project/o2r-finder/issues/13

nuest commented:
Thanks @LukasLohoff !

nuest commented:
@jansule do you use any other fields in the queries already, see https://github.com/o2r-project/o2r-web-api/pull/57/files#diff-a0415b49c9410c15a8694fed28aa36aeR71 ?

LukasLohoff commented on docs/search.md:
````diff
@@ -1,42 +1,160 @@
 ```
 
-## Suggesters - WORK IN PROGRESS
+#### Spatial search
````
The geo_shape field has to be updated in this example to match the metadata: `metadata.o2r.spatial.union.geojson.geometry`

This query worked for me: 
```
    "query": {
        "bool": {
            "must": {
                "match_all": {}
            },
            "filter": {
                "geo_shape": {
                    "metadata.o2r.spatial.union.geojson.geometry": {
                        "shape": {
                            "type": "polygon",
                            "coordinates":
                                [
                                    [
                                        [
                                            -7.294921874999999,
                                            38.54816542304656
                                        ],
                                        [
                                            34.013671875,
                                            38.54816542304656
                                        ],
                                        [
                                            34.013671875,
                                            68.6245436634471
                                        ],
                                        [
                                            -7.294921874999999,
                                            68.6245436634471
                                        ],
                                        [
                                            -7.294921874999999,
                                            38.54816542304656
                                        ]
                                    ]
                                ]
                        },
                        "relation": "within"
                    }
                }
            }
        }
    }
```

LukasLohoff replied:
@nuest Additionally I had some issues getting the spatial search to work:

The GeoJSON "coordinates" array in the generated compendium had a wrong structure:
- the array wasn't nested deep enough, it must be e.g. `[ [ [1,2], [2,3], [3,4], [1,2] ] ]` for a polygon
- the last point didn't equal the first point (a rectangular polygon needs 5 points to be valid)

Maybe this was because I generated the test compendia a while ago and it could be fixed already.

nuest replied:
Yes, I assume the coordinate generation from the extractor works now, see https://github.com/o2r-project/o2r-meta/issues?utf8=%E2%9C%93&q=is%3Aissue%20geojson%20
